### PR TITLE
fix(cli): skip printing summary when output mode is set to json/yaml (issue: #18620)

### LINKED ISSUE #18620: app sync --output json produces non-json
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

We are trying to add a verification CI pipeline to every pull request against our argocd "gitops" repo.  This where we have argocd deploy itself.

One of the validation is to ensure any potential syncs a pull request may trigger is valid prior to merging the code.  We make use of the `--dry-run` option of `argocd app sync`.

To make this programmatic, we want the output of the `--dry-run` to produce json, but unfortunately the output is mixed with non-json text.

**To Reproduce**

`argocd` is the name of our app
`argoproj/argocd/dev` is the path to the helm chart and values.

```shell
argocd app sync --dry-run --output json argocd --local argoproj/argocd/dev | jq .
```

**Expected behavior**

the output would be valid json that can be parsed by `jq` without error

**Version**

```shell
argocd: v2.10.7+b060053.dirty
  BuildDate: 2024-04-15T12:31:39Z
  GitCommit: b060053b099b4c81c1e635839a309c9c8c1863e9
  GitTreeState: dirty
  GoVersion: go1.22.2
  Compiler: gc
  Platform: darwin/arm64
argocd-server: v2.8.0+804d4b8```

### PR DESCRIPTION
**Pull request by Fluder-Paradyne**
Fixes  #18620

The reported issue also occurs when the output is set to YAML while using the sync, wait, and rollback commands.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by pasha-codefresh** [comment]
Could you add tests for this change?

**Comment by Fluder-Paradyne** [comment]
Working on it, will mention once I am done

**Comment by Fluder-Paradyne** [comment]
@pasha-codefresh Hi, I have added the test.
please review.

File: cmd/argocd/commands/app_test.go [hunk 1]
```diff
@@ -1849,6 +1858,168 @@ func testApp(name, project string, labels map[string]string, annotations map[str
 	}
 }
 
+func TestWaitOnApplicationStatus_JSONandYAMLOutput(t *testing.T) {
```
Comment (pasha-codefresh) [inline cmd/argocd/commands/app_test.go:1861]: Could you please also cover scenario when no output flag is provided.

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

File: cmd/argocd/commands/app_test.go (hunk 1 above)
Comment (Fluder-Paradyne) [reply cmd/argocd/commands/app_test.go]: have added the test for it,
please check when you are free

**Review by pasha-codefresh — APPROVED** [review]

**Pull request merged**